com.unity.ml-agents/Runtime/TeamManagerIdCounter.cs

```diff
@@ -0,0 +1,13 @@
+using System.Threading;
+
+namespace Unity.MLAgents
+{
+    internal static class TeamManagerIdCounter
+    {
+        static int s_Counter;
+        public static int GetTeamManagerId()
+        {
+            return Interlocked.Increment(ref s_Counter); ;
```

Comment:
Could we make this class more generic and have Agent also use this for their AgentId?

Reply:
AgentId (or actually Episode ID) uses EpisodeIdCounter which is almost the same thing. It's a static class so I figured I'd make a separate class. Should I try to integrate them?